### PR TITLE
Feat: signMNIST 분류 인공지능 모델 구현

### PR DESCRIPTION
**Pull request by YujinJI**
## 변경사항
- Train_signMNIST.ipynb 파일이 추가되었습니다.
> Denoising_AI.ipynb 파일은.. 작업한 뒤에 올리겠습니다.

### 정수로 이루어져있던 `label`을 알파벳으로 매핑 후, `LabelEncoder`를 통해 정수로 매핑해주었습니다.
- 기존에는 0부터 25까지(`A`부터 `Z`까지) 이루어져 있었지만, 위 과정을 통해 `J`와 `Z`가 없음에 따라 0부터 23까지로 수정되었습니다.
- 따라서 `y_train = utils.to_categorical(y_train, NUM_CLASSES)`를 문제없이 진행할 수 있었습니다.

### 모델링 아직도 익숙하지 않습니다. 🙄
- 스터디때는 어떤 코드를 참조해왔는지.. `x = layers.BatchNormalization()(x)` `x = layers.LeakyReLU()(x)`를 사용하고 있었습니다. 주제랑은 맞지 않는 것 같아 지웠지만 잘 지운건지 모르겠네욥.
- 모델링 부분은 강사님의 코드를 참고해서 수정했습니다.
- Functional API로 모델을 만들었습니다.

### CODE REVIEW

**Comment by BluePinetree** [comment]
_Commented on notebook `week5/지유진/Train_signMNIST.ipynb` Cell 15 Line 10_
```python

alphabet = string.ascii_uppercase

y_train = [alphabet[label] for label in y_train if label < len(alphabet)]
y_train = pd.Series(y_train)

y_test = [alphabet[label] for label in y_test if label < len(alphabet)]
y_test = pd.Series(y_test)
```
존재하지 않는 레이블을 제외하고 벡터화를 진행하기위해 클래스 index를 알파벳으로 돌려놓는 과정이군여. string 라이브러리에서 가져오는 방법은 저도 처음보는 경우인데 감사링. 좋슴다

_Respond and view the context [here](https://gitnotebooks.com/SPAI-Lab/GenerativeAI_Study/pull/24#comment-1069)._

**Comment by BluePinetree** [comment]
_Commented on notebook `week5/지유진/Train_signMNIST.ipynb` Cell 22 Line 9_
```python

# LabelEncoder를 사용하여 문자열을 정수로 매핑
label_encoder = LabelEncoder()
y_train = label_encoder.fit_transform(y_train)
y_train = pd.Series(y_train)

y_test = label_encoder.fit_transform(y_test)
y_test = pd.Series(y_test)
```
이전의 과정에서 수행한 일종의 index to label을 하는 과정 뒤, `scikit-learn` 라이브러리의  `LabelEncoder`를 사용하여 label to index를 진행하였습니다. 위와 같은 과정으로 포함되지 않는 레이블인 `J`와 `Z`에 대한 index는 제외하고 새롭게 레이블링이 가능합니다. 멋집니다.

추가적으로, 만약 위의 과정을 `LabelEncoder`를 사용하지 않고 수행하려면, 다음과 같이 코드를 작성해도 좋을 듯 합니다.

```python
import string

alphabet = list(string.ascii_uppercase)
CLASS_LABELS = [c for c in string.ascii_uppercase if not c == 'J' and not c == 'Z']
label_to_idx = {c:i for i,c in enumerate(CLASS_LABELS)}
idx_to_label = {i:c for i,c in enumerate(CLASS_LABELS)}

train_y = np.array([label_to_idx[alphabet[idx]] for idx in train_y])
test_y = np.array([label_to_idx[alphabet[idx]] for idx in test_y])
```

_Respond and view the context [here](https://gitnotebooks.com/SPAI-Lab/GenerativeAI_Study/pull/24#comment-1070)._

**Comment by BluePinetree** [comment]
_Commented on notebook `week5/지유진/Train_signMNIST.ipynb` Cell 27 Line 19_
```python

x = layers.Conv2D(filters=32, kernel_size=3, strides=1, padding="same", activation="relu")(x)
x = layers.MaxPooling2D(pool_size=2, strides=2)(x)

x = layers.Flatten()(x)

x = layers.Dense(48, activation="relu")(x)
x = layers.Dense(NUM_CLASSES)(x)
```
출력층의 특징의 개수를 `NUM_CLASSES` 변수로 대체하면서 범주(category)의 수가 변해도 대응할 수 있는 모델이 되었습니다.

_Respond and view the context [here](https://gitnotebooks.com/SPAI-Lab/GenerativeAI_Study/pull/24#comment-1071)._

**Review by BluePinetree — APPROVED** [review]
더욱 편리한 레이블링 방법을 사용하였으며, 훈련 및 평가까지 문제없이 깔끔하게 진행되었습니다. Good.

* 모델링이 익숙하지 않은 부분은 걱정하지 않아도 됩니다. 익숙해질때까지 반복하다보면 감이오게 될겁니다. Keep Going.
